src/runtime_src/core/tools/xbutil2/SubCmdStatus.cpp

```diff
@@ -0,0 +1,605 @@
+{
+  uint32_t count = 0;
+  for(uint64_t i = 0; i < map->m_count; i++) {
+    if(map->m_debug_ip_data[i].m_type != type)
+      continue;
+
+    if (baseAddress)
+      baseAddress->push_back(map->m_debug_ip_data[i].m_base_address);
+    if(portNames) {
+      std::string portName;
+      // Fill up string with 128 characters (padded with null characters)
+      portName.assign(map->m_debug_ip_data[i].m_name, 128);
+      // Strip away any extraneous null characters
+      portName.assign(portName.c_str());
+      portNames->push_back(portName);
+    }
+    ++count;
+  }
+  return count;
+}
+
+
+std::pair<size_t, size_t> getCUNamePortName (std::vector<std::string>& aSlotNames,
+                             std::vector< std::pair<std::string, std::string> >& aCUNamePortNames) 
+{
+  //Slotnames are of the format "/cuname/portname" or "cuname/portname", split them and return in separate vector
+  //return max length of the cuname and port names
+  size_t max1 = 0, max2 = 0;
+  char sep = '/';
+  for (auto slotName: aSlotNames) {
+    size_t found1;
+    size_t start = 0;
+    found1 = slotName.find(sep, 0);
+    if (found1 == 0) {
+      //if the cuname starts with a '/'
+      start = 1;
+      found1 = slotName.find(sep, 1);
+    }
+    if (found1 != std::string::npos) {
+      aCUNamePortNames.emplace_back(slotName.substr(start, found1-start), slotName.substr(found1+1));
+    }
+    else {
+      aCUNamePortNames.emplace_back("Unknown", "Unknown");
+    }
+    //Replace the name of the host-AIM to something simple
+    if (aCUNamePortNames.back().first.find("interconnect_host_aximm") != std::string::npos) {
+      aCUNamePortNames.pop_back();
+      aCUNamePortNames.emplace_back("XDMA", "N/A");
+    }
+    // Use strlen() instead of length() because the strings taken from debug_ip_layout
+    // are always 128 in length, where the end is full of null characters
+    max1 = std::max(strlen(aCUNamePortNames.back().first.c_str()), max1);
+    max2 = std::max(strlen(aCUNamePortNames.back().second.c_str()), max2);
+  }
+  return std::pair<size_t, size_t>(max1, max2);
+}
+
+std::pair<size_t, size_t> getStreamName (const std::vector<std::string>& aSlotNames,
+                             std::vector< std::pair<std::string, std::string> >& aStreamNames) 
+{
+  //Slotnames are of the format "Master-Slave", split them and return in separate vector
+  //return max length of the Master and Slave port names
+  size_t max1 = 0, max2 = 0;
+  for (auto &s: aSlotNames) {
+    size_t found;
+    found = s.find(IP_LAYOUT_SEP, 0);
+    if (found != std::string::npos)
+      aStreamNames.emplace_back(s.substr(0, found), s.substr(found+1));
+    else
+      aStreamNames.emplace_back("Unknown", "Unknown");
+    max1 = std::max(aStreamNames.back().first.length(), max1);
+    max2 = std::max(aStreamNames.back().second.length(), max2);
+  }
+  return std::pair<size_t, size_t>(max1, max2);
+}
+
+
+int readAIMCounters(xclDeviceHandle hdl, debug_ip_layout* map) 
```

Comment:
If this is meant to be an error code, then that is not allowed, you should use exceptions and be a 'void' function.
I see though that this function always returns 0 so no exceptions, but the return code is meaningless.

Reply:
Done.